.github/workflows/pr.yml

```diff
@@ -22,3 +28,66 @@ jobs:
         ref: main
         token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
         inputs: '{ "context":"${{ env.CONTEXT }}", "actor":"${{ github.actor }}", "repository":"${{ github.repository }}", "branch":"${{ env.BRANCH }}", "sha":"${{ env.SHA }}", "token":"${{ secrets.ELEVATED_GITHUB_TOKEN }}" }'
+
+  get-go-version:
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "go-version=$(cat .go-version)" >> "${GITHUB_OUTPUT}"
+
+  linting:
+    name: golangci-lint
+    needs:
+      - get-go-version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Setup go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+          cache: false
+
+      - name: Setup GOROOT  # Need to set GOROOT because an older version of go-critic used GOROOT to find rules
+        run: echo "GOROOT=$(go env GOROOT)" >> "${GITHUB_ENV}"
+
+      - name: golangci-lint-helm-gen
```

Comment:
Let me know if you want teach of these in their own runner... I think it's okay running as a single job?

Reply:
I would put in its own lint.yaml or copy what the Consul repo has for naming.